core/src/main/scala/dagr/core/execsystem/GraphNode.scala

```diff
@@ -37,11 +39,10 @@ import scala.collection.mutable.ListBuffer
   *                         is used to track submission, start, and end dates for the parent node.  Typically parent nodes
   *                         are [[dagr.core.tasksystem.Pipeline]]s.
   */
-class GraphNode(val taskId: BigInt,
-                var task: Task,
+class GraphNode(var task: Task,
                 predecessorNodes: Traversable[GraphNode] = Nil,
                 var state: GraphNodeState.Value = GraphNodeState.PREDECESSORS_AND_UNEXPANDED,
-                val parent: Option[GraphNode] = None) {
+                val parent: Option[GraphNode] = None) extends BaseGraphNode {
 
   private val _predecessors = new ListBuffer[GraphNode]()
   private val _predecessorsStatic = new ListBuffer[GraphNode]()
```

Comment:
This caught my eye even though you didn't change it.  It's odd that this is called `static` but is a mutable type, and is in fact mutated in the add methods.  I looked around and it seems like the only real usage outside of this class is in one test.  I wonder if it's even worth keeping around in that case?